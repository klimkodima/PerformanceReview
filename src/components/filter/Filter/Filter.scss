.filter {
  padding: 15px;

  &__header {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 40px;
  }

  &__title {
    color: #818181;
    margin: 0;
    font-weight: 400;
    font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
    font-size: 1rem;
    letter-spacing: 0.00938em;
  }

  .clear-button {
    color: #9f9f9f;
    padding: 20px;
    font-size: 1rem;
    line-height: 0;
  }
}
