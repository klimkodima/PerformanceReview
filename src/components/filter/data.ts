export const FILTERS_NAMES = {
  TEAM: 'TEAM',
  LEVEL: 'LEVEL',
  AUDITOR: 'AUDITOR'
};

export const mockTeamFilter: string[] = ['Alpha', 'Charlie', 'X-Rays'];

export const mockLevelsFilter: string[] = [
  'Trainee',
  'Junior',
  'Middle',
  'Senior',
  'Team Lead'
];

export const mockAuditorsFilter: string[] = [
  'Alex Vagner',
  'Tom Cruise',
  'Fred Jhonson',
  'Tawes',
  'Kerry',
  'Jim',
  'Kevin',
  'Fred',
  'Franklin',
  'Ted',
  'Johnson',
  'Bob'
];
