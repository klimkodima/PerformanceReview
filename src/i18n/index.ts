import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import common from './locales.en.json';

const resources = {
  en: {
    translation: common
  }
};

const setI18nextConfig = async () =>
  await i18n.use(LanguageDetector).use(initReactI18next).init({
    resources,
    fallbackLng: 'en'
  });

void setI18nextConfig();

export default i18n;
