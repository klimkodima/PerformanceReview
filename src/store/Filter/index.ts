export {
  selectCheckedTeams,
  selectCheckedAuditors,
  selectCheckedAuditorsData,
  selectIsDisabledReset,
  selectTeamsData,
  selectFilteredTeams
} from './selectors';
export type { FilterDataType, AuditorType } from './types';
export { FilterSaga } from './sagas';
export {
  default as filterReducer,
  fetchFilterData,
  filterByAuditor,
  checkResetEnabled,
  filterByLevel,
  filterByTeam,
  reset,
  searchForAuditors,
  setFiltersData,
  setFilteredAuditorsData,
  resetFilterUsersData
} from './reducer';
