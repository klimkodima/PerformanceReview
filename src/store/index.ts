import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import rootReducer from './rootReducer';
import rootSaga from './rootSaga';

const sagaMiddleware = createSagaMiddleware();

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['dateFilter', 'tasksCoefficient']
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
      }
    }).concat(sagaMiddleware)
});

sagaMiddleware.run(rootSaga);

export type RootStateType = ReturnType<typeof rootReducer>;
export type StoreType = typeof store;

export const persistor = persistStore(store);
export default store;
