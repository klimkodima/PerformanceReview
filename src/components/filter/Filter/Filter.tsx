import { ReactElement, memo } from 'react';

import { useDispatch } from 'react-redux';

import { useTranslation } from 'react-i18next';

import { Button } from '@mui/material';

import { FilterList } from '../FilterList';

import { reset } from 'src/store/Filter/reducer';
import { useTypedSelector } from '../hooks/useTypedSelector';

import { FilterPropsType } from '../types';

import './Filter.scss';

const Filter = memo(
  ({
    levelFilter,
    teamFilter,
    auditorFilter
  }: FilterPropsType): ReactElement => {
    const { t } = useTranslation();
    const {
      filteredTeams,
      filteredLevels,
      filteredAuditors,
      checkedTeams,
      checkedLevels,
      checkedAuditors,
      isDisabledReset
    } = useTypedSelector((state) => state.filter);

    const dispatch = useDispatch();

    return (
      <div className='filter'>
        <div className='filter__wrapper'>
          <div className='filter__header'>
            <h3 className='filter__title'>{t('filter.filters_title_text')}</h3>
            <Button
              className='clear-button'
              onClick={() => dispatch(reset())}
              disabled={isDisabledReset}
            >
              {t('filter.clear_button_text')}
            </Button>
          </div>
          <div>
            <FilterList
              title={t('filter.team')}
              filter={teamFilter}
              filteredData={filteredTeams}
              checkedItems={checkedTeams}
            />
            <FilterList
              title={t('filter.level')}
              filter={levelFilter}
              filteredData={filteredLevels}
              checkedItems={checkedLevels}
            />
            <FilterList
              title={t('filter.auditor')}
              filter={auditorFilter}
              filteredData={filteredAuditors}
              checkedItems={checkedAuditors}
            />
          </div>
        </div>
      </div>
    );
  }
);

export default Filter;

Filter.displayName = 'Filter';
