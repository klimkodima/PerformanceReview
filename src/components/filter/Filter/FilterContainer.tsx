import { ReactElement, useEffect } from 'react';

import { useDispatch } from 'react-redux';

import { checkResetEnabled, setFilteredAuditorsData } from 'src/store/Filter';
import Filter from './Filter';
import { useTypedSelector } from '../hooks/useTypedSelector';
import { getAuditors } from '../helpers/functions';

import { AuditorType } from 'src/store/Filter/types';

const FilterContainer = (): ReactElement => {
  const dispatch = useDispatch();
  const { data, searchedAuditors, isShowFilter } = useTypedSelector(
    (state) => state.filter
  );

  const teams: string[] = Array.from(
    new Set(data.map(({ team }: AuditorType) => team))
  );

  const levels: string[] = Array.from(
    new Set(data.map(({ level }: AuditorType) => level))
  );

  const auditors = getAuditors(searchedAuditors, data);

  useEffect(() => {
    dispatch(checkResetEnabled());
    dispatch(setFilteredAuditorsData());
  }, [data]);

  if (isShowFilter) {
    return (
      <Filter
        teamFilter={teams}
        levelFilter={levels}
        auditorFilter={auditors}
      />
    );
  } else {
    return <></>;
  }
};

export default FilterContainer;
