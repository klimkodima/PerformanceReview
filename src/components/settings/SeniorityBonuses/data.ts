import { nanoid } from 'nanoid';

import { BonusesDataType } from 'src/store/SeniorityBonuses';

export const mockBonusesData: BonusesDataType[] = [
  {
    id: nanoid(),
    experience: {
      value: '12',
      isError: false,
      errorText: ''
    },
    bonus: {
      value: '10',
      isError: false,
      errorText: ''
    }
  },
  {
    id: nanoid(),
    experience: {
      value: '15',
      isError: false,
      errorText: ''
    },
    bonus: {
      value: '25',
      isError: false,
      errorText: ''
    }
  },
  {
    id: nanoid(),
    experience: {
      value: '25',
      isError: false,
      errorText: ''
    },
    bonus: {
      value: '30',
      isError: false,
      errorText: ''
    }
  },
  {
    id: nanoid(),
    experience: {
      value: '30',
      isError: false,
      errorText: ''
    },
    bonus: {
      value: '35',
      isError: false,
      errorText: ''
    }
  }
];
