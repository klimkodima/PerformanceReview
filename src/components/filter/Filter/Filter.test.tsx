import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { I18nextProvider } from 'react-i18next';
import i18n from 'src/i18n';
import Filter from './Filter';
import { mockTeamFilter, mockLevelsFilter, mockAuditorsFilter } from '../data';
import store from '../../../store';

describe('Filter', () => {
  test('Matches snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <Filter
              teamFilter={mockTeamFilter}
              levelFilter={mockLevelsFilter}
              auditorFilter={mockAuditorsFilter}
            />
          </I18nextProvider>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
