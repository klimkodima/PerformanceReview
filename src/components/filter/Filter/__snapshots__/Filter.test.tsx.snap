// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Matches snapshot 1`] = `
<div
  className="filter"
>
  <div
    className="filter__wrapper"
  >
    <div
      className="filter__header"
    >
      <h3
        className="filter__title"
      >
        FILTERS
      </h3>
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled clear-button css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        CLEAR
      </button>
    </div>
    <div>
      <div
        className="filter-list"
      >
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          data-testid="filter-list-test"
        >
          <div
            className="filter-list__name"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
            >
              TEAM
            </p>
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium plus-button filter-list-button-TEAM css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              +
            </button>
          </div>
        </ul>
      </div>
      <div
        className="filter-list"
      >
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          data-testid="filter-list-test"
        >
          <div
            className="filter-list__name"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
            >
              LEVEL
            </p>
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium plus-button filter-list-button-LEVEL css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              +
            </button>
          </div>
        </ul>
      </div>
      <div
        className="filter-list"
      >
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          data-testid="filter-list-test"
        >
          <div
            className="filter-list__name"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
            >
              AUDITOR
            </p>
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium plus-button filter-list-button-AUDITOR css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              +
            </button>
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
